src/mjolnir/ingest_transit.cc

```diff
@@ -143,98 +129,104 @@ struct unique_transit_t {
-std::priority_queue<tileTransitInfo> select_transit_tiles(const boost::property_tree::ptree& pt) {
+std::priority_queue<tile_transit_info_t> select_transit_tiles(const boost::property_tree::ptree& pt) {
 
   auto path = pt.get<std::string>("mjolnir.transit_feeds_dir");
 
   std::set<GraphId> tiles;
-  const auto& tile_level = TileHierarchy::levels().back();
-  std::unordered_map<GraphId, tileTransitInfo> tile_map;
+  const auto& local_tiles = TileHierarchy::levels().back().tiles;
+  std::unordered_map<GraphId, tile_transit_info_t> tile_map;
+
+  // adds a tile_info for the tile_id to tile_map if there is none yet, and returns the tile_info
+  // object
+  auto get_tile_info = [&tile_map, &local_tiles](const gtfs::Stop& stop) -> tile_transit_info_t& {
+    auto graphid = GraphId(local_tiles.TileId(stop.stop_lat, stop.stop_lon),
+                           TileHierarchy::GetTransitLevel().level, 0);
+
+    return tile_map.insert({graphid, tile_transit_info_t{graphid}}).first->second;
+  };
 
   filesystem::recursive_directory_iterator gtfs_feed_itr(path);
   filesystem::recursive_directory_iterator end_file_itr;
   for (; gtfs_feed_itr != end_file_itr; ++gtfs_feed_itr) {
     if (filesystem::is_directory(gtfs_feed_itr->path())) {
       auto feed_path = gtfs_feed_itr->path().string();
-      LOG_INFO("Trying to load " + feed_path);
+      LOG_INFO("Loading " + feed_path);
       gtfs::Feed feed(feed_path);
       feed.read_feed();
-      LOG_INFO("Done Reading Feed");
+
       const auto& stops = feed.get_stops();
+      // 1st pass to add all the stations, so we can add stops to its children in a 2nd pass
+      for (const auto& stop : stops) {
+        if (stop.location_type == gtfs::StopLocationType::Station) {
+          auto& tile_info = get_tile_info(stop);
+          tile_info.stations.insert({stop.stop_id, feed_path});
+        }
+      }
 
+      // 2nd pass to add the platforms/stops
       for (const auto& stop : stops) {
-        if (stop.stop_id.empty()) {
+        // TODO: GenericNode & BoardingArea could be useful at some point
+        if (!(stop.location_type == gtfs::StopLocationType::StopOrPlatform) &&
+            !(stop.location_type == gtfs::StopLocationType::EntranceExit)) {
           continue;
         }
-        gtfs::Id currStopId = stop.stop_id;
-        float x = stop.stop_lon;
-        float y = stop.stop_lat;
 
-        GraphId currId =
-            GraphId(tile_level.tiles.TileId(y, x), TileHierarchy::GetTransitLevel().level, 0);
+        auto& tile_info = get_tile_info(stop);
 
-        tileTransitInfo currTile;
-
-        if (tile_map.find(currId) != tile_map.end()) {
-          currTile = tile_map[currId];
+        // if this station doesn't exist, we need to create it: we use the fact that this entry is
+        // not a station type to fake a station object in write_stops()
+        auto station_in_tile = tile_info.stations.find({stop.parent_station, feed_path});
```

Comment:
the `parent_station` may be empty for stop/platform types, aparently. what this means is that this code will presumeably still work (because empty string wont occur as a real id for another station) but it would be clearer if we said something about it or updated the if to check for empty or something

Reply:
ah yeah, exactly here do distinguish already, albeit not in the most obvious manner..